sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -0,0 +1,328 @@
+  }
+
+  def mergeStatementNotExistWhenClauseError(ctx: MergeIntoTableContext): Throwable = {
+    new ParseException("There must be at least one WHEN clause in a MERGE statement", ctx)
+  }
+
+  def nonLastMatchedClauseOmitConditionError(ctx: MergeIntoTableContext): Throwable = {
+    new ParseException("When there are more than one MATCHED clauses in a MERGE " +
+      "statement, only the last MATCHED clause can omit the condition.", ctx)
+  }
+
+  def nonLastNotMatchedClauseOmitConditionError(ctx: MergeIntoTableContext): Throwable = {
+    new ParseException("When there are more than one NOT MATCHED clauses in a MERGE " +
+      "statement, only the last NOT MATCHED clause can omit the condition.", ctx)
+  }
+
+  def emptyPartitionKeyError(key: String, ctx: PartitionSpecContext): Throwable = {
+    new ParseException(s"Found an empty partition key '$key'.", ctx)
+  }
+
+  def combinationQueryResultClausesUnsupportedError(ctx: QueryOrganizationContext): Throwable = {
+    new ParseException(
+      "Combination of ORDER BY/SORT BY/DISTRIBUTE BY/CLUSTER BY is not supported", ctx)
+  }
+
+  def distributeByUnsupportedError(ctx: QueryOrganizationContext): Throwable = {
+    new ParseException("DISTRIBUTE BY is not supported", ctx)
+  }
+
+  def transformWithSerdeUnsupportedError(ctx: ParserRuleContext): Throwable = {
+    new ParseException("TRANSFORM with serde is only supported in hive mode", ctx)
+  }
+
+  def lateralWithPivotInFromClauseNotAllowedError(ctx: FromClauseContext): Throwable = {
+    new ParseException("LATERAL cannot be used together with PIVOT in FROM clause", ctx)
+  }
+
+  def windowRepetitiveError(name: String, ctx: WindowClauseContext): Throwable = {
+    new ParseException(s"The definition of window '$name' is repetitive", ctx)
+  }
+
+  def invalidWindowReferenceError(name: String, ctx: WindowClauseContext): Throwable = {
+    new ParseException(s"Window reference '$name' is not a window specification", ctx)
+  }
+
+  def cannotResolveWindowReferenceError(name: String, ctx: WindowClauseContext): Throwable = {
+    new ParseException(s"Cannot resolve window reference '$name'", ctx)
+  }
+
+  def joinCriteriaUnimplementedError(join: JoinCriteriaContext, ctx: RelationContext): Throwable = {
+    new ParseException(s"Unimplemented joinCriteria: $join", ctx)
+  }
+
+  def naturalCrossJoinUnsupportedError(ctx: RelationContext): Throwable = {
+    new ParseException("NATURAL CROSS JOIN is not supported", ctx)
+  }
+
+  def emptyInputForTableSampleError(ctx: ParserRuleContext): Throwable = {
+    new ParseException("TABLESAMPLE does not accept empty inputs.", ctx)
+  }
+
+  val byteLengthLiteralMsg = "byteLengthLiteral"
+
+  val bucketOutOfColumnMsg = "BUCKET x OUT OF y ON colname"
+
+  val bucketOutOfFunctionMsg = "BUCKET x OUT OF y ON function"
+
+  def tableSampleByBytesUnsupportedError(msg: String, ctx: SampleMethodContext): Throwable = {
+    new ParseException(s"TABLESAMPLE($msg) is not supported", ctx)
+  }
```

Comment:
Instead of defining error messages as vals that will not be reusable, I would prefer directly specify these messages when throwing exceptions. For example 
```scala
throw QueryParsingErrors.tableSampleByBytesUnsupportedError("byteLengthLiteral", ctx)
```

Reply:
OK